### PR TITLE
Remove the "native" suffix from the codebase

### LINKED ISSUE #14186: [RFC] Rename the fsdp strategy to FairscaleFSDP since we have fsdp native now.
Retrospectively, we should have named the fsdp strategy FairscaleFSDP or something like that. Normally, everything else we have always implies that it is "native" anyway.

_Originally posted by @awaelchli in https://github.com/Lightning-AI/lightning/pull/14092#pullrequestreview-1070992407_

cc @borda @awaelchli @carmocca @rohitgr7 @akihironitta

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

See title. This is consistent with Fabric

Fixes #14186 

### Does your PR introduce any breaking changes? If yes, please list them.

- `strategy="fsdp_native"` -> `strategy="fsdp"`
- (pl) `strategy="fsdp_native_full_shard_offload"` -> `strategy="fsdp_cpu_offload"`
- (fabric) `strategy="fsdp_full_shard_offload"` -> `strategy="fsdp_cpu_offload"`
- `pytorch_lightning.strategies.fully_sharded_native.DDPFullyShardedNativeStrategy` -> `pytorch_lightning.strategies.fsdp.FSDPStrategy`
- `pytorch_lightning.plugins.precision.fsdp_native_native_amp.FullyShardedNativeNativeMixedPrecisionPlugin` -> `pytorch_lightning.plugins.precision.fsdp.FSDPMixedPrecisionPlugin`
- `pytorch_lightning.plugins.precision.native_amp` -> `pytorch_lightning.plugins.precision.amp`
- `lightning_fabric.plugins.precision.native_amp` -> `lightning_fabric.plugins.precision.amp`
- `NativeSyncBatchNorm` -> `TorchSyncBatchNorm`

cc @justusschock @awaelchli @borda @carmocca

### CODE REVIEW

File: tests/tests_pytorch/strategies/test_fsdp.py [hunk 1]
```diff
@@ -11,8 +11,8 @@
 from pytorch_lightning import Trainer
 from pytorch_lightning.callbacks import ModelCheckpoint
 from pytorch_lightning.demos.boring_classes import BoringModel
-from pytorch_lightning.plugins.precision.fsdp_native_native_amp import FullyShardedNativeNativeMixedPrecisionPlugin
-from pytorch_lightning.strategies import DDPFullyShardedNativeStrategy
+from pytorch_lightning.plugins.precision.fsdp_amp import FSDPMixedPrecisionPlugin
```
Comment (awaelchli) [inline tests/tests_pytorch/strategies/test_fsdp.py:14]: fyi, in fabric the module is called fsdp

File: src/lightning_fabric/plugins/precision/fsdp.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 import torch
 from typing_extensions import Literal
 
-from lightning_fabric.plugins.precision.native_amp import MixedPrecision
```
Comment (awaelchli) [inline src/lightning_fabric/plugins/precision/fsdp.py:19]: Could you also consider removing the prefix from fabrics native_amp module?

**Review by awaelchli — APPROVED** [review]

File: src/lightning_fabric/plugins/precision/fsdp.py (hunk 2 above)
Comment (carmocca) [reply src/lightning_fabric/plugins/precision/fsdp.py:19]: I did this already. Can you point out what I missed exactly?

File: tests/tests_pytorch/strategies/test_fsdp.py (hunk 1 above)
Comment (carmocca) [reply tests/tests_pytorch/strategies/test_fsdp.py:14]: Good catch. Changed

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Pull request merged**